### DISCUSSION TITLE
Anonymous mode on separate port

### DISCUSSION THREAD

**Discussion by jaskij** [grafana/grafana#66026] [General]
# Overview
---

An option to have Grafana respond on multiple ports, only one of which allows anonymous access.

# Why? (The use case)
---

We are deploying Grafana on an embedded device allowing both network access and a kiosk. For the kiosk itself, we would strongly prefer allowing anonymous access, as this ensures no issues with logging in or out happen. As network access will be exposed to site's internal network, it requires login. Also, removing session timeouts or similar would be ill-advised from a security standpoint.

See also https://github.com/grafana/grafana/issues/2729

# How
---

My initial idea was to allow limiting anonymous access by IP ranges (eg only allow `127.0.0.0/24`).

But, after thinking further, I realized that there is a simpler way to implement it: Grafana could expose two TCP ports, of which only one allows anonymous access. Then whatever tools are between it and the network (a reverse proxy or a firewall) could handle the rest.

Whether this secondary port also allows logging in and regular access is something I am ambivalent about and could perhaps be another toggle.

# Other options
---

This could *probably* be somehow achieved with an advanced configuration in the reverse proxy, perhaps requiring two similar dashboards. But it's a convoluted option which requires duplication and a lot of effort.

Another option would probably be to run separate instances, but this still runs into the issue of duplicating the dashboard.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82777. Feel free to continue the discussion around this there. Thank you!
